platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ConsensusImpl.java

```diff
@@ -16,1037 +16,751 @@
-    //
-    // If it is a witness, then it is also stored in exactly
-    // one of the RoundInfo.witnesses lists, for the appropriate round.
-    // If it is, then it might also be in the RoundInfo.famousWitnesses
-    // for that same round.
-    ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
-
+    private final List<EventImpl> recentEvents = new LinkedList<>();
     /** stores all round information */
     private final ConsensusRounds rounds;
-
     /**
-     * Number of events that have reached consensus order. This is used for setting consensus order numbers
-     * in events, so it must be part of the signed state.
+     * Number of events that have reached consensus order. This is used for setting consensus order
+     * numbers in events, so it must be part of the signed state.
      */
-    private volatile long numConsensus = 0;
+    private long numConsensus = 0;
 
     /**
-     * The minimum consensus timestamp for the next event that reaches consensus. This is null if no event
-     * has reached consensus yet. As each event reaches its consensus, its timestamp is moved forward (if
-     * necessary) to be at least this time.
-     * And then minTimestamp is moved forward by n * {@link EventImpl#MIN_TRANS_TIMESTAMP_INCR_NANOS} nanoseconds, if
-     * the
-     * event had n transactions (or n=1 if no transactions).
-     * Then minTimestamp is rounded up to the nearest multiple of {@link EventImpl#MIN_TRANS_TIMESTAMP_INCR_NANOS}
+     * The last consensus timestamp. This is equal to the consensus time of the last transaction in
+     * the last event that reached consensus. This is null if no event has reached consensus yet.
+     * As each event reaches its consensus, its timestamp is moved forward (if necessary) to be
+     * after this time by n {@link ConsensusConstants#MIN_TRANS_TIMESTAMP_INCR_NANOS} nanoseconds,
+     * if the event had n transactions (or n=1 if no transactions).
      */
-    private Instant minTimestamp = null;
-
-    /** an event with this number is "marked", all others are "unmarked". Used by the ValidAncestorsIterator */
-    private int currMark = 1;
-
-    /** the round number passed to setAddressBook the last time it was called. The next call must be 1 greater */
-    private long prevRoundSetAddressBook;
-
-    /** the number of coin rounds that have happened so far (used to update the statistics) */
-    private long numCoinRounds = 0;
-
-    ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
-    // Public constructors
-    ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+    private Instant lastConsensusTime = null;
+    /**
+     * if consensus is not starting from genesis, this instance is used to accurately calculate the
+     * round for events
+     */
+    private InitJudges initJudges;
```

Comment:
initialize variable to null?   (Java does it automatically, but you initialize variables to null in other locations.)

Reply:
done